### ISSUE TITLE
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) unable to open database file on startup

### ISSUE THREAD

**Issue by TheLux83** [Technosoft2000/docker-calibre-web#25]
Hey there,

I'm trying to get my calibre-web running on a Ubuntu Server 16.04.4 LTS Host.
Unfortunately I don't know why, but it isn't able to run without problems.

Here is the log-output.
```
[INFO] Docker image version: 1.1.10,
[INFO] Alpine Linux version: 3.6.2,
[INFO] Create group calibre with id 1000,
[INFO] Create user calibre with id 1000,
[INFO] Current active timezone is UTC,
Wed Mar 28 10:53:56 CEST 2018,
[INFO] Container timezone is changed to: Europe/Berlin,
[INFO] Change the ownership of /calibre-web (including subfolders) to calibre:calibre,
[INFO] Current git version is:,
git version 2.13.5,
[INFO] Checkout the latest Calibre-Web version ...,
[INFO] ... git clone -b master --single-branch https://github.com/janeczku/calibre-web.git /calibre-web/app -v,
Cloning into '/calibre-web/app'...,
POST git-upload-pack (189 bytes),
[INFO] Autoupdate is active, try to pull the latest sources for Calibre-Web ...,
[INFO] ... current git status is,
On branch master,
Your branch is up-to-date with 'origin/master'.,
nothing to commit, working tree clean,
d01d7b91a8be744bf35e820795709c97ab7c8654,
[INFO] ... pulling sources,
Already up-to-date.,
[INFO] ... git status after update is,
[INFO] Config directory option is DEACTIVATED,
> due this the Calibre books directory /books will be used to store the configuration,
[INFO] Checking permissions of the config directory: /books,
> Output is: 755 calibre 1000 calibre 1000,
> Permissions: 755,
> Assigned group: calibre,
> Assigned group ID: 1000,
> Assigned owner: calibre,
> Assigned owner ID: 1000,
> Using permissions for checks: 0755,
> The user calibre:1000 is the owner and has write access at /books,
[INFO] 'app.db' and 'gdrive.db' will be linked into /books,
> create 'app.db' link /calibre-web/app/app.db assigned to source /books/app.db,
> create 'gdrive.db' link /calibre-web/app/gdrive.db assigned to source /books/gdrive.db,
[INFO] Checking permissions of the books directory: /books,
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) unable to open database file,
    return self.dbapi.connect(*cargs, **cparams),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 393, in connect,
    return dialect.connect(*cargs, **cparams),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/strategies.py", line 105, in connect,
    connection = pool._invoke_creator(self),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 651, in __connect,
    self.__connect(first_connect_check=True),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 461, in __init__,
    return _ConnectionRecord(self),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 333, in _create_connection,
    return self._create_connection(),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 1229, in _do_get,
    rec = pool._do_get(),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 516, in checkout,
    fairy = _ConnectionRecord.checkout(pool),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 766, in _checkout,
    return _ConnectionFairy._checkout(self),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 387, in connect,
    return fn(),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 2147, in _wrap_pool_connect,
    reraise(type(exception), exception, tb=exc_tb, cause=cause),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/util/compat.py", line 203, in raise_from_cause,

    reraise(type(exception), exception, tb=exc_tb, cause=cause),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/base.py", line 2147, in _wrap_pool_connect,
    return fn(),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 387, in connect,
    return _ConnectionFairy._checkout(self),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 766, in _checkout,
    fairy = _ConnectionRecord.checkout(pool),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 516, in checkout,
    rec = pool._do_get(),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 1229, in _do_get,
    return self._create_connection(),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 333, in _create_connection,
    return _ConnectionRecord(self),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 461, in __init__,
    self.__connect(first_connect_check=True),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/pool.py", line 651, in __connect,
    connection = pool._invoke_creator(self),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/strategies.py", line 105, in connect,
    return dialect.connect(*cargs, **cparams),
  File "/usr/lib/python2.7/site-packages/sqlalchemy/engine/default.py", line 393, in connect,
    return self.dbapi.connect(*cargs, **cparams),
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) unable to open database file
```

I know, that we had a similar problem here: https://github.com/Technosoft2000/docker-calibre-web/issues/2
But I have a metadata.db file in my volume.

Here is my docker-compose file:
```
calibre-web:
    image: technosoft2000/calibre-web
    network_mode: "bridge"
    container_name: calibre-web
    ports:
      - "8083:8083"
    volumes:
      - /media/data/eBooks:/books
    environment:
      - SET_CONTAINER_TIMEZONE=true
      - CONTAINER_TIMEZONE=Europe/Berlin
      - PUID=1000
      - PGID=1000
      - VIRTUAL_HOST={Virtual Host}
      - VIRTUAL_PORT=8083
      - LETSENCRYPT_HOST={LetsEncrypt Host}
      - LETSENCRYPT_EMAIL={LetsEncrypt Mail}
```

The share /media/data comes from my Windows-Server but the UID and GID were configured to 1000, which is my User on my Ubuntu Server and I'm able to create and delete files and folders.

I've checked my eBook Folder and I can see an app.db but no gdrive.db file.

Can anyone help me? I don't know what to do.

**Comment by Technosoft2000** [comment]
Hi @TheLux83 

I've fixed in the latest release v1.1.11 the ownership of the symlinks of _app.db_, _gdrive.db_ and _kindlegen_ which is changed from _root_ to the _calibre_ user & group and maybe this helps in your case.

**Comment by Technosoft2000** [comment]
@arnie311 

could you please post the parameters which you've used for the creation of the container and also the detailed information (UID and GID) of the user which owns the mapped _/calibre-web/config_ directory

For example if your user is named **docker** then execute the command `id docker` to get the detailed information - like:
```
id docker
uid=1029(docker) gid=100(users) groups=100(users),65539(docker)
```
